plugins/alerting/pkg/alerting/api.go

```diff
@@ -3,17 +3,22 @@ package alerting
 import (
 	"context"
 
-	"github.com/rancher/opni/pkg/alerting/backend"
+	"github.com/rancher/opni/pkg/alerting/drivers/backend"
+	"github.com/rancher/opni/pkg/alerting/shared"
 	alertingv1 "github.com/rancher/opni/pkg/apis/alerting/v1"
+	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	"github.com/samber/lo"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 // --- Trigger ---
-
 func (p *Plugin) TriggerAlerts(ctx context.Context, req *alertingv1.TriggerAlertsRequest) (*alertingv1.TriggerAlertsResponse, error) {
 	lg := p.Logger.With("Handler", "TriggerAlerts")
 	lg.Debugf("Received request to trigger alerts  on condition %s", req.GetConditionId())
 	lg.Debugf("Received alert annotations : %s", req.Annotations)
-
+	if err := req.Validate(); err != nil {
```

Comment:
Should move this check above the log statements

Reply:
Yep